drivers/gpio/gpio_renesas_ra8.c

```diff
@@ -0,0 +1,244 @@
+/*
+ * Copyright (c) 2024 Renesas Electronics Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT renesas_ra8_gpio
+
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/pinctrl.h>
+#include <zephyr/drivers/gpio/gpio_utils.h>
+#include <zephyr/irq.h>
+#include <soc.h>
+
+#define PRCR_KEY   0xA500U
+#define VBATT_PORT 4
+
+static const gpio_pin_t vbatt_pins[] = {
+	2U,
+	3U,
+	4U,
+};
+
+#define RA8_INVALID_PORT_ADDR 0UL
+
+#define RA8_GPIO_PORT_ADDR(nodelabel) \
+	COND_CODE_1(DT_NODE_EXISTS(DT_NODELABEL(nodelabel)), \
+	(DT_REG_ADDR(DT_NODELABEL(nodelabel))), (RA8_INVALID_PORT_ADDR))
+
+struct gpio_ra8_config {
+	struct gpio_driver_config common;
+	uint8_t port_num;
+	R_PORT0_Type *port;
+};
+
+struct gpio_ra8_data {
+	struct gpio_driver_data common;
+};
+
+static int gpio_ra8_pin_configure(const struct device *dev, gpio_pin_t pin, gpio_flags_t flags)
+{
+	const struct gpio_ra8_config *config = dev->config;
+
+	if (((flags & GPIO_INPUT) != 0U) && ((flags & GPIO_OUTPUT) != 0U)) {
+		return -ENOTSUP;
+	}
+
+	if ((flags & GPIO_PULL_DOWN) != 0U) {
+		return -ENOTSUP;
+	}
+
+	if (config->port_num == VBATT_PORT) {
+		uint32_t clear = 0;
+
+		for (int i = 0; i < ARRAY_SIZE(vbatt_pins); i++) {
+			if (vbatt_pins[i] == pin) {
+				WRITE_BIT(clear, i, 1);
+			}
+		}
+
+		R_SYSTEM->PRCR = ((R_SYSTEM->PRCR | PRCR_KEY) | R_SYSTEM_PRCR_PRC1_Msk);
+
+		R_SYSTEM->VBTICTLR &= (uint8_t)~clear;
+
+		R_SYSTEM->PRCR = ((R_SYSTEM->PRCR | PRCR_KEY) & (uint16_t)~R_SYSTEM_PRCR_PRC1_Msk);
+	}
+
+#if BSP_CFG_MCU_PART_SERIES == 8
```

Comment:
Why are we not using a Zephyr definition like SOC_SERIES_RA8?

Reply:
@ydamigos : Thank you for your feedback. We will change to use Zephyr macro. However, we found that this is redundant codes in this file which support for RA8 only. We removed these redundant codes. Please check. Thank you.